interfaces/thirdparty/cp2k.py

```diff
@@ -140,25 +142,72 @@ def get_runtime(self):
         diff = endTime - startTime
         return diff.total_seconds()
 
-    def _get_energy_type(self, search='Total', index=0):
+    def _get_energy_type(self, search='Total', index=-1, unit='a.u.'):
         if index:
             select = index
         else:
-            select = -1
-        s = self.grep_output(search + ' energy:')[select].split()[-1]
-        return float(s)
+            select = slice(0,None)
 
-    def get_energy(self, index=0):
-        """Return last occurence of 'Total energy:' in the output."""
-        return self._get_energy_type('Total', index=index)
+        s = self.grep_output(search+' energy:')[select]
+        if isinstance(select, int):
```

Comment:
```suggestion
        if not isinstance(select, slice):
```

I'd recommend swapping things around here.

In its current state the check will fail with, for example, the likes of `numpy.integer` or any other object which defines a `__index__` method.